metricshub-jdbc-extension/src/main/java/org/sentrysoftware/metricshub/extension/jdbc/JdbcConfiguration.java

```diff
@@ -0,0 +1,208 @@
+package org.sentrysoftware.metricshub.extension.jdbc;
+
+/*-
+ * ╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲
+ * MetricsHub JDBC Extension
+ * ჻჻჻჻჻჻
+ * Copyright 2023 - 2024 Sentry Software
+ * ჻჻჻჻჻჻
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * ╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱
+ */
+
+import static com.fasterxml.jackson.annotation.Nulls.SKIP;
+
+import com.fasterxml.jackson.annotation.JsonSetter;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Builder.Default;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.sentrysoftware.metricshub.engine.common.exception.InvalidConfigurationException;
+import org.sentrysoftware.metricshub.engine.common.helpers.StringHelper;
+import org.sentrysoftware.metricshub.engine.configuration.IConfiguration;
+import org.sentrysoftware.metricshub.engine.deserialization.TimeDeserializer;
+
+/**
+ * Represents JDBC configuration in the MetricsHub extension system.
+ */
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class JdbcConfiguration implements IConfiguration {
+
+	private static final int DEFAULT_POSTGRESQL_PORT = 5432;
+	private static final int DEFAULT_MYSQL_PORT = 3306;
+	private static final int DEFAULT_ORACLE_PORT = 1521;
+	private static final int DEFAULT_MSSQL_PORT = 1433;
+	private static final int DEFAULT_INFORMIX_PORT = 9088;
+	private static final int DEFAULT_H2_PORT = 9092;
+	private static final int INVALID_PORT = -1;
+
+	private String username;
+	private char[] password;
+	private char[] url;
+
+	@Default
+	@JsonSetter(nulls = SKIP)
+	@JsonDeserialize(using = TimeDeserializer.class)
+	private final Long timeout = 120L;
+
+	private String type;
+
+	private Integer port;
+	private String database;
+
+	private String hostname;
+
+	@Override
+	public String toString() {
+		return username != null ? "JDBC as " + username : "JDBC";
+	}
+
+	@Override
+	public void validateConfiguration(String resourceKey) throws InvalidConfigurationException {
+		StringHelper.validateConfigurationAttribute(
+			username,
+			attr -> attr == null || attr.isBlank(),
+			() ->
+				String.format(
+					"Resource %s - No username configured for JDBC." +
```

Comment:
The string is currently not multiline. If you need it to be, we can use Java 17's multiline string feature with `"""`.

Reply:
Yeah, it's just that it could be now:

```java
() -> """
  Resource %s - No username configured for JDBC.\
  This resource will not be monitored.\
  Please verify the configured username value.
""".formatted(resourceKey);
```

Reply:
Yep, maybe with an additional `\` to skip the last `\n`:
```java
```java
() -> """
  Resource %s - No username configured for JDBC.\
  This resource will not be monitored.\
  Please verify the configured username value.\
""".formatted(resourceKey);
```

Reply:
Or 
```java
() -> """
  Resource %s - No username configured for JDBC.\
  This resource will not be monitored.\
  Please verify the configured username value.""".formatted(resourceKey);
```